### PR TITLE
Set CARGO_TARGET_DIR in nix dev shell

### LINKED ISSUE #751: Complete rebuild of dependencies occurs too often with vscode
At first a complete rebuild occurred *every* time I did anything in a terminal. This problem has been alleviated by @nyospe suggestion to add the following to `.vscode/settings.json`:
```
"rust-analyzer.server.extraEnv": {
        "CARGO_TARGET_DIR": "target_dirs/vsc_ra"
    },
```
Despite this, I can still trigger a complete rebuild just by running `cargo test` in the terminal. Nathan's doc also suggests to add `export CARGO_TARGET_DIR='target_dirs/nix` to `flake.nix`. But `flake.nix` is complex and I don't know where to add this.

### PR DESCRIPTION
**Pull request by sveitser**
@ggutoski can you confirm if this helps before we merge?

Because I think it will prevent the `cargo-sweep` tool from finding the target directory when it runs recursively from outside this repo :-/

Close #751

### CODE REVIEW

**Comment by sveitser** [comment]
Also need to fix a few things on the CI for this.

File: .github/workflows/build_static.yml [hunk 1]
```diff
@@ -30,13 +30,13 @@ jobs:
...
         run: |
-          echo "TARGET=x86_64-unknown-linux-musl" >> $GITHUB_ENV
+          echo "TARGET_TRIPLET=x86_64-unknown-linux-musl" >> $GITHUB_ENV
           echo "DEVSHELL=.#crossShell" >> $GITHUB_ENV
 
       - name: Prepare ARM
         if: ${{ matrix.arch == 'arm' }}
         run: |
-          echo "TARGET=aarch64-unknown-linux-musl" >> $GITHUB_ENV
+          echo "TARGET_TRIPLET=aarch64-unknown-linux-musl" >> $GITHUB_ENV
```
Comment (ggutoski) [inline .github/workflows/build_static.yml:39]: Do you really mean `aarch64` here?
```suggestion
          echo "TARGET_TRIPLET=arch64-unknown-linux-musl" >> $GITHUB_ENV
```

**Review by ggutoski — APPROVED** [review]
lol I don't even know why I'm reviewing this. 😛 One minor comment to address at your discretion.

File: .github/workflows/build_static.yml (hunk 1 above)
Comment (sveitser) [reply .github/workflows/build_static.yml:39]: The first `a` is for `Arm` I think: https://en.wikipedia.org/wiki/AArch64

**Comment by sveitser** [comment]
The remaining CI failure is because we're not adding the correct directory to the path now that we use a different target directory for nix. Will fix.

**Comment by sveitser** [comment]
> lol I don't even know why I'm reviewing this. 😛 One minor comment to address at your discretion.

Thanks!

**Pull request merged**